### PR TITLE
Group by Actor Class

### PR DESCRIPTION
**Pull request by mfitton**
## Why are these changes needed?
We're moving towards a version of the logical view where, by default, aggregate statistics are displayed for each type of actor (mapping to a Python class on the back-end). This PR introduces first steps, including the code to group by actor class and expandable panels per-actor with the title of the actor. In the next PR, these expandable panels will begin un-expanded and display summary stats. The overall goal of these changes is to make the logical view more navigable and less overwhelming when there are a large number of actors.

Screen Shot 2020-08-17 at 9 22 47 AM
Screen Shot 2020-08-17 at 9 22 58 AM
Screen Shot 2020-08-17 at 9 23 11 AM

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/dashboard/client/src/api.ts [hunk 1]
```diff
@@ -187,21 +186,21 @@ export type FullActorInfo = {
 };
 
 export type PartialActorInfo = {
-  actorId: string;
-  actorTitle: string;
+  actorId?: string;
```
Comment (rkooo567) [inline python/ray/dashboard/client/src/api.ts:189]: When are they supposed to be null?

File: python/ray/dashboard/node_stats.py [hunk 2]
```diff
@@ -91,6 +92,55 @@ def get_node_stats(self):
...
+                for worker_info in workers_info:
+                    if "coreWorkerStats" in worker_info:
+                        core_worker_stats = worker_info["coreWorkerStats"]
+                        addr = (core_worker_stats["ipAddress"],
+                                str(core_worker_stats["port"]))
+                        if addr in self._addr_to_actor_id:
+                            actor_info = actors[self._addr_to_actor_id[addr]]
+                            format_reply_id(core_worker_stats)
+                            actor_info.update(core_worker_stats)
+                            actor_info["averageTaskExecutionSpeed"] = round(
```
Comment (rkooo567) [inline python/ray/dashboard/node_stats.py:115]: Isn't this ignoring the fact that there is time when actors are idle?

**Review by rkooo567 — COMMENTED** [review]
Code in general LGTM, but I have more of high level questions. 

1. Does this mean we won't show the hierarchical relation of actors anymore? Or is group by actor class optional?
1. If we don't remove hierarchical relation of actors, how do we handle the case where the same actor class created by different parents?

File: python/ray/dashboard/client/src/api.ts (hunk 1 above)
Comment (mfitton) [reply python/ray/dashboard/client/src/api.ts:189]: They seem to be null all of the time for these `PartialActorInfo` which are actually pending Actor tasks. I've been relying on the mapping that the backend passes from actorId to actor info for `PartialActorInfo` instances.

File: python/ray/dashboard/node_stats.py (hunk 2 above)
Comment (mfitton) [reply python/ray/dashboard/node_stats.py:115]: Yes, it is, you're right. However, this is the same behavior we currently have on the dashboard. Maybe we should brainstorm a way to capture only the time when the actor is not idle.

**Comment by mfitton** [comment]
Good questions @rkooo567 
It does remove the hierarchical actor rendering. I'm slowly working towards this spec: https://docs.google.com/document/d/14c4Sw8G9wc_diXBnyUz1tPFh896iuWLr9saBW0Gl-Jc/edit#heading=h.u4av5dyeaceu
However, I do think the hierarchical actor information can be helpful at times. It might be worth keeping the hierarchical option as a choice. I can introduce a checkbox or something similar, at least for the time being.
@robertnishihara do you think it's worth keeping the hierarchical view around?

File: python/ray/dashboard/node_stats.py (hunk 2 above)
Comment (rkooo567) [reply python/ray/dashboard/node_stats.py:115]: Yeah that sounds good to me!

File: python/ray/dashboard/client/src/api.ts (hunk 1 above)
Comment (rkooo567) [reply python/ray/dashboard/client/src/api.ts:189]: I see. I think it is probably possible to fix it (in a way we can show Pending actors' ID. This should be possible now after we enabled GCS actor management). For the scope of this PR it LGTM.

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Let's have another PR if we'd like to have hierarchy stuff back again!

**Comment by robertnishihara** [comment]
Nice start :) Looking forward to seeing the unexpanded version :)

**Pull request merged**